chasm/field.go

```diff
@@ -113,25 +120,40 @@ func (f Field[T]) Get(chasmContext Context) (T, error) {
 				err = serviceerror.NewInternalf("pointer field referenced an unhandled value: %v", referencedNode.fieldType())
 			}
 			if err != nil {
-				return nilT, err
+				// nolint:forbidigo // Panic is intended here for framework error handling.
+				panic(err)
 			}
 			nodeValue = referencedNode.value
 		}
 	case fieldTypeDeferredPointer:
 		// For deferred pointers, return the component directly stored in v
 		nodeValue = f.Internal.v
 	default:
-		return nilT, serviceerror.NewInternalf("unsupported field type: %v", f.Internal.fieldType())
+		// nolint:forbidigo // Panic is intended here for framework error handling.
+		panic(serviceerror.NewInternalf("unsupported field type: %v", f.Internal.fieldType()))
 	}
 
 	if nodeValue == nil {
-		return nilT, nil
+		return nilT, false
 	}
 	vT, isT := nodeValue.(T)
 	if !isT {
-		return nilT, serviceerror.NewInternalf("node value doesn't implement %s", reflect.TypeFor[T]().Name())
+		// nolint:forbidigo // Panic is intended here for framework error handling.
+		panic(serviceerror.NewInternalf("node value doesn't implement %s", reflect.TypeFor[T]().Name()))
 	}
-	return vT, nil
+	return vT, true
+}
+
+// Get returns the value of the field, deserializing it if necessary.
+// Panics rather than returning an error, as errors are supposed to be handled by the framework as opposed to the
+// application.
+func (f Field[T]) Get(chasmContext Context) T {
+	v, ok := f.TryGet(chasmContext)
+	if !ok {
+		// nolint:forbidigo // Panic is intended here for framework error handling.
+		panic(serviceerror.NewInternalf("field value of type %s not found", reflect.TypeFor[T]().Name()))
```

Comment:
nit: not sure if we want to mention the "of type %s" part, which should already panic in the type assertion in TryGet(). I think here it just means this method is called on an empty field or due to some bugs in the framework, value can't find/ is not populated.